File: java/ankit/arrays/InsertionSort.java
// link- https://workat.tech/problem-solving/practice/implement-insertion-sort

class Solution {
	void insertionSort (int[] arr) {
		// add your logic here
		for(int i=1;i<arr.length;i++){
			int j=i-1;
			int key=arr[i];
			while(j>=0 && arr[j]>key){
				arr[j+1]=arr[j];
				j--;
			}
			arr[j+1]=key;
			
		}
	}
}

File: python/vedant_bhatnagar/InsertionSort.py
class Solution:
	def insertionSort(self, arr: List[int]) -> List[int]:
		# add your logic here
		if(len(arr) == 1):
			return arr
		
		n = len(arr)
		
		for wall in range(1, n):
			curr = arr[wall]
			pos = wall-1
			
			while pos>=0 and arr[pos]>curr:
				arr[pos+1] = arr[pos]
				pos = pos-1
			arr[pos+1] = curr
		return arr
				



